x-pack/legacy/plugins/code/server/queue/index_worker.ts

```diff
@@ -211,18 +188,97 @@ export class IndexWorker extends AbstractWorker {
   private getProgressReporter(
     repoUri: RepositoryUri,
     revision: string,
-    index: number,
-    total: number
+    type: IndexerType,
+    total: number // total number of indexers
   ) {
     return async (progress: IndexProgress) => {
+      const indexStatus: IndexWorkerProgress = await this.objectClient.getRepositoryIndexStatus(
+        repoUri
+      );
       const p: IndexWorkerProgress = {
         uri: repoUri,
-        progress: progress.percentage,
+        progress: indexStatus.progress,
         timestamp: new Date(),
         revision,
-        indexProgress: progress,
       };
-      return await this.objectClient.setRepositoryLspIndexStatus(repoUri, p);
+
+      switch (type) {
+        case IndexerType.COMMIT:
+          p.commitIndexProgress = progress;
+          p.progress =
+            progress.percentage +
+            (indexStatus.indexProgress ? indexStatus.indexProgress.percentage : 0);
+          break;
+        case IndexerType.LSP:
+          p.indexProgress = progress;
+          p.progress =
+            progress.percentage +
+            (indexStatus.commitIndexProgress ? indexStatus.commitIndexProgress.percentage : 0);
+          break;
+        default:
+          this.log.warn(`Unknown indexer type ${type} for indexing progress report.`);
+          break;
+      }
+      return await this.objectClient.updateRepositoryIndexStatus(repoUri, p);
+    };
+  }
+
+  private async shouldJobResume(
+    payload: any
+  ): Promise<{
+    resume: boolean;
+    checkpointReqs: Map<IndexerType, IndexRequest | undefined>;
```

Comment:
checkpointReqs parameter needs to be documented, this function feel little bitt hard to read because of the use of this map

Reply:
add some comment inline. so basically, this is just an internal function to refactor/wrap code to make the main function of this worker cleaner and shorter. The map here contains all the possible checkpoint requests of different type of indexers since we introduced a new kind (CommitIndexer).